src/http/web_socket.cr

```diff
@@ -90,10 +90,11 @@ class HTTP::WebSocket
     end
   end
 
-  def close(message = nil)
+  # Closes the websocket with *reason* and *code*.
+  def close(reason : String? = nil, code : Int16? = nil)
```

Comment:
I think `code` should be the first parameter. This matches the way its encoded in the payload, and lots of services I've dealt with more often than not will send *only* `code`.

You also cannot send *just* reason, so I suggest:

```cr
def close
  # ..
end

def close(code : Int16, reason : String? = nil)
  # ..
end
```
See RFC 6455: https://tools.ietf.org/html/rfc6455#section-5.5.1

>    The Close frame MAY contain a body (the "Application data" portion of
   the frame) that indicates a reason for closing, such as an endpoint
   shutting down, an endpoint having received a frame too large, or an
   endpoint having received a frame that does not conform to the format
   expected by the endpoint.  **If there is a body, the first two bytes of
   the body MUST be a 2-byte unsigned integer** (in network byte order)
   representing a status code with value /code/ defined in Section 7.4.
   Following the 2-byte integer, **the body MAY contain UTF-8-encoded data
   with value /reason/, the interpretation of which is not defined by
   this specification.**  This data is not necessarily human readable but
   may be useful for debugging or passing information relevant to the
   script that opened the connection.

Reply:
Well, the context of the comment doesn't explain it in any way, so I'll repeat my question: why split the `#close` method into 2 implementations?

Reply:
The RFC excerpt above lays it out.

> The Close frame MAY contain a body

So we have two methods:

1. `def close` - this sends no body (`Bytes.empty`), which will be interpreted implicitly by the server/client as close 1005 [RFC](https://tools.ietf.org/html/rfc6455#section-7.4.1)

2. `def close(code, reason = nil)` - Sends a body. When you send a body, `code` is required, and `reason` is optional.

This captures the two behaviors of section 5.5.1.

Reply:
@z64 Implementation on the Crystal side have nothing to do with RFC(s).

IMO it should be simply defined as:

```cr
def close(code : UInt16? = nil, reason : String? = nil)
  # ...
end
```

Reply:
The problem with that is this allows a user to do

`ws.close(reason: "foo")`

This is invalid per ws protocol. You *must* supply a code. Having `code : UInt16` gives us compile time safety instead of a runtime check.

Reply:
@z64 Oops, I missed that part, thanks for bearing with me! :) If that's the case, then implementation enforcing it during compile-time is definitely a good thing 👍

Reply:
@vladfaust Please, don't _resolve_ the conversation until is actually implemented.